### PR TITLE
[MNT] Set up pre-commit

### LINKED ISSUE #1: Set up `pre-commit`
Install and configure `pre-commit` hooks to make code style consistent.
Pre-commit hooks to install:
- [x] Flake 8
- [x] Black
- [x] Mypy

Pre-commit docs: https://pre-commit.com/index.html
Pre-commit support hooks: https://pre-commit.com/hooks.html
Example config file: https://github.com/neurobagel/api/blob/main/.pre-commit-config.yaml

### PR DESCRIPTION
**Pull request by Raya679**
- Closes #1

Changes proposed in this pull request:
  Pre-commit hooks installed:
  - [x] Flake 8
  - [x] Black
  - [x] Mypy

## Checklist
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [x] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  - repo: https://github.com/PyCQA/flake8
+    rev: 7.0.0
+    hooks:
+      - id: flake8
+        language_version: python3
+        args:
+          - --extend-ignore=E501
+          - --extend-select=B950
+
```
Comment (rmanaem) [inline .pre-commit-config.yaml]: I'm assuming these were copied over from the API repo so I'd suggest removing these args for now as they're specific to the API code. We can always add them back in if needed.

**Review by rmanaem — CHANGES_REQUESTED** [review]
Hey @Raya679, thanks for the PR! :tada:

I left a comment that requires a change please take a look.
:cherries: I think it makes sense to add a requirements.txt file with `pre-commit` to make setting up an environment and installing pre-commit hooks easier.

Lastly, I took the liberty of updating your PR description with our PR template.
Please take a look and add an appropriate prefix to the PR title.
For more details on the PR template refer to this [page](https://neurobagel.org/contributing/CONTRIBUTING/) of our doc.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Raya679) [reply .pre-commit-config.yaml]: Hii @rmanaem,
I have made the required changes. Please let me know if any more changes are required.
Thanks!

**Review by rmanaem — APPROVED** [review]
Thanks for implementing the changes @Raya679! Feel free to squash and merge :cook:
P.S. I suggest using the [MNT] prefix here since we use [CI] mainly for workflows.

**Pull request merged**